lib/SILOptimizer/Mandatory/DataflowDiagnostics.cpp

```diff
@@ -46,16 +46,17 @@ static void diagnoseMissingReturn(const UnreachableInst *UI,
     llvm_unreachable("unhandled case in MissingReturn");
   }
 
-  // No action required if the function returns 'Void' or that the
-  // function is marked 'noreturn'.
-  if (ResTy->isVoid() || F->isNoReturnFunction())
+  // No action required if the function returns 'Void'.
+  if (ResTy->isVoid())
```

Comment:
Is even this necessary? SILGen ought to emit the return implicitly already if it's emitting a void function. We'd still want to diagnose things like `guard` or `switch` exhaustiveness where we use unreachables to indicate control flow that should be unreachable.